code/game/machinery/_machinery.dm

```diff
@@ -517,7 +517,7 @@
 	var/is_dextrous = FALSE
 	if(isanimal(user))
 		var/mob/living/simple_animal/user_as_animal = user
-		if (user_as_animal.dextrous)
+		if (ISADVANCEDTOOLUSER(user_as_animal))
```

Comment:
Make this check a different trait if they're not an advanced tool user, this one's already bloated and I do *not* want to see what else happens by giving carp this.